### PR TITLE
[4.0] Namespace associations fields

### PR DESCRIPTION
**Pull request by laoneo**
Namespace associations fields.

### CODE REVIEW

File: administrator/components/com_associations/forms/association.xml [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 		<field
 			name="modalassociation"
-			type="modalassociation"
+			type="modal_association"
```
Comment (wilsonge) [inline administrator/components/com_associations/forms/association.xml]: Does this need backporting to staging as a bug? or is this an upgrade issue that needs to be documented?

Comment (laoneo) [inline administrator/components/com_associations/forms/association.xml]: In staging the class name is [`JFormFieldModalAssociation`](https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_associations/models/fields/modalassociation.php) which is correct. The rest of Joomla has this kind of form fields in a `modal` subfolder and then is the type `modal_association`with the underscore. I just changed it that it is like the rest of the core extensions.

Comment (wilsonge) [reply administrator/components/com_associations/forms/association.xml:15]: So we're making a b/c break that needs documentation to standardise this field to be similar to all the other modal fields right?

Comment (laoneo) [reply administrator/components/com_associations/forms/association.xml:15]: Correct

**Pull request merged**